src/pages/workspaces/workspace/workflows/LaunchAnalysisModal.js

```diff
@@ -99,31 +128,23 @@ export default ajaxCaller(class LaunchAnalysisModal extends Component {
     } else if (processSingle) {
       this.launch()
     } else if (type === EntitySelectionType.processAll) {
-      this.setState({ message: 'Fetching data...' })
-      const entities = _.map('name', await Workspaces.workspace(namespace, name).entitiesOfType(rootEntityType))
       this.createSetAndLaunch(entities)
     } else if (type === EntitySelectionType.chooseRows) {
-      const entities = _.keys(selectedEntities)
-      if (_.size(entities) === 1) {
-        this.launch(rootEntityType, _.head(entities))
+      if (entities.length === 1) {
+        this.launch(rootEntityType, entities[0])
       } else {
         this.createSetAndLaunch(entities)
       }
     } else if (type === EntitySelectionType.processMergedSet) {
-      if (_.size(selectedEntities) === 1) {
-        const { entityType, name } = _.values(selectedEntities)[0]
+      if (entities.length === 1) {
+        const { entityType, name } = entities[0]
         this.launch(entityType, name, `this.${rootEntityType}s`)
       } else {
-        const entities = _.flow(
-          _.flatMap(`attributes.${rootEntityType}s.items`),
-          _.map('entityName')
-        )(selectedEntities)
-
         this.createSetAndLaunch(entities)
       }
     } else if (type === EntitySelectionType.chooseSets) {
-      if (_.size(selectedEntities) === 1) {
-        this.launch(rootEntityType, _.values(selectedEntities)[0].name)
+      if (_.size(entities) === 1) {
+        this.launch(rootEntityType, _.values(entities)[0].name)
```

Comment:
TOL: It's confusing to me that `entities` has a different data type depending on what mode we're in.

Reply:
This is true. I think it's a candidate for further simplification later.